network/server.go

```diff
@@ -303,7 +326,15 @@ func (s *Server) OnTraffic(conn *ConnWrapper, stopConnection chan struct{}) Acti
 	go func(server *Server, conn *ConnWrapper, stopConnection chan struct{}, stack *Stack) {
 		for {
 			server.Logger.Trace().Msg("Passing through traffic from client to server")
-			if err := server.Proxy.PassThroughToServer(conn, stack); err != nil {
+
+			// Find the proxy associated with the given connection
+			proxy, exists := server.GetProxyForConnection(conn)
+			if !exists {
+				server.Logger.Error().Msg("Failed to find proxy that matches the connection")
+				break
```

Comment:
Should we close the connection at this point? Just curious.

Reply:
The 'not exist' error is kinda panic error, so yes, probably we should close it. I’ve added `stopConnection <- struct{}{} `right after that.